src/components/headerSearch/headerSearch.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState } from "react";
+import Mark from "mark.js";
+import './headerSearch.css';
+import { IonButton, IonButtons, IonIcon, IonInput, IonItemDivider } from "@ionic/react";
+import { arrowDown, arrowUp } from 'ionicons/icons';
+
+interface Props {
+  doc: any
+}
+
+const HeaderSearch: React.FC<Props> = ({doc}) => {
+  const [needle, setNeedle] = useState<string>('');
+  const [marks, setMarks] = useState<NodeListOf<HTMLElement>>(document.querySelectorAll('mark'));
+  const [currentIndex, setCurrentIndex] =  useState<number>(0);
+  let mark: any;
```

Comment:
This is a functional component, so `mark` will be null with every render.

Reply:
I prefer class components so that you can stash non-state variables and not lose them.